test/transfer-sol.ts

```diff
@@ -0,0 +1,121 @@
+const { expect } = require("chai");
+import { BigNumber } from "@ethersproject/bignumber";
+import { Result, RLP } from "ethers/lib/utils";
+import { decodeExit, encodeExit } from "../ts/coders";
+import { transfer } from "../ts/transfer";
+import { Exit } from "../ts/types";
+const { ethers } = require("hardhat");
+import { TestConsumer } from "../typechain/TestConsumer";
+
+describe("transfer (solidity)", function () {
+  let testConsumer: TestConsumer;
+
+  before(async () => {
+    const exitFormat = await (
+      await ethers.getContractFactory("ExitFormat")
+    ).deploy();
+
+    await exitFormat.deployed();
+
+    testConsumer = await (
+      await ethers.getContractFactory("TestConsumer", {
+        libraries: {
+          ExitFormat: exitFormat.address,
+        },
+      })
+    ).deploy();
+
+    await testConsumer.deployed();
+  });
+
+  it("Can transfer", async function () {
+    const initialOutcome: Exit = [
+      {
+        asset: "0x0000000000000000000000000000000000000000",
+        data: "0x",
+        allocations: [
+          {
+            destination: "0x96f7123E3A80C9813eF50213ADEd0e4511CB820f",
+            amount: "0x05",
+            data: "0x",
+          },
+          {
+            destination: "0x53484E75151D07FfD885159d4CF014B874cd2810",
+            amount: "0x05",
+            data: "0x",
+          },
+        ],
+      },
+    ];
+
+    const initialHoldings = [BigNumber.from(6)];
+    const indices = [[1]];
+
+    const {
+      updatedHoldings,
+      updatedOutcome,
+      exit,
+    } = await testConsumer.transfer(initialOutcome, initialHoldings, indices);
+
+    const gasEstimate = await testConsumer.estimateGas.transfer(
+      initialOutcome,
+      initialHoldings,
+      indices
+    );
+
+    expect(gasEstimate.toNumber()).to.equal(44518);
```

Comment:
This is an alternative pattern to recording gas consumption (that I quite like). 

Instead of recording the consumption in some CI artifact file, we can run the test locally and adjust the assertion until the test passes. Then, the gas consumption is stored in the test file itself.

Reply:
This is great in so far as it automates us becoming consciously aware of gas changes. 

Weren't there some non-deterministic calculations in other tests? (Probably via a different calculation method than this - via ganache instead of the `estimateGas` function).

Reply:
Yes, this number comes from hardhat's[ test environment. ](https://hardhat.org/hardhat-network/). I don't know if it is more reliable than ganache or not (it uses some of the same tech under the hood).